### PR TITLE
feature(query): fix read quoted string

### PR DESCRIPTION
**Pull request by sundy-li**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

## Changelog

- Bug Fix

## Related Issues

Fixes #issue

### CODE REVIEW

**Comment by b41sh** [comment]
we should also add some unit tests for `test_parse_value_source`

**Comment by sundy-li** [comment]
@ZeaLoVe   Seems the back quoted logic is incorrect in logic test.
```
AssertionError: Expected:
1 1 1
2 2 \"2\"-\"2\"
 Actual:
        1        1        1
        2        2  "2"-"2"
 Statement:
Parsed Statement
    at_line: 15,
    s_type: Statement: query, type: IIT, query_type: IIT, label: ['mysql', 'http'], retry: False,
    suite_name: gen/09_fuse_engine/09_0001_remote_insert,
    text:
        SELECT * FROM t1;
    results: [(<re.Match object; span=(0, 10), match='---- mysql'>, 21, '1 1 1\n2 2 "2"-"2"'), (<re.Match object; span=(0, 9), match='---- http'>, 25, '1 1 1\n2 2 \\"2\\"-\\"2\\"')],
```

File: common/io/src/buffer/buffer_read_ext.rs [hunk 1]
```diff
@@ -46,8 +46,42 @@ pub trait BufferReadExt: BufferRead {
 
     fn read_quoted_text(&mut self, buf: &mut Vec<u8>, quota: u8) -> Result<()> {
         self.must_ignore_byte(quota)?;
-        self.keep_read(buf, |b| b != quota)?;
-        self.must_ignore_byte(quota)
+
+        loop {
```
Comment (zhang2014) [inline common/io/src/buffer/buffer_read_ext.rs:50]: CSV format also need this.

**Review by zhang2014 — APPROVED** [review]

**Comment by ZeaLoVe** [comment]
cc @youngsofun , seams http handler return 1 when expect bool

**Comment by ZeaLoVe** [comment]
https://github.com/datafuselabs/databend/blob/main/tests/logictest/suites/gen/09_fuse_engine/09_0001_remote_insert#L25   this case need change into:

statement query IIT
SELECT * FROM t1;

---- 
1 1 1
2 2 "2"-"2"

**Comment by youngsofun** [comment]
log

> handler return 1 when expect bool

return `"true"` on main on my mac

and logic test can pass 

```
(venv) ➜  logictest git:(fix) curl --request POST '127.0.0.1:8001/v1/query/' --header 'Content-Type: application/json' --data-raw '{"sql": "select true", "string_fields":true}' -u root:

{"id":"d601cd71-7a8e-4e31-862b-283ca2b422d9","session_id":"ea35c032-3b36-4cc4-a4b8-30445b9417d3","schema":{"fields":[{"name":"true","default_expr":null,"data_type":{"type":"Boolean"}}],"metadata":{}},"data":[["true"]],"state":"Succeeded","error":null,"stats":{"scan_progress":{"rows":1,"bytes":1},"running_time_ms":7.6482909999999995},"stats_uri":"/v1/query/d601cd71-7a8e-4e31-862b-283ca2b422d9","final_uri":"/v1/query/d601cd71-7a8e-4e31-862b-283ca2b422d9/final","next_uri":null,"kill_uri":"/v1/query/d601cd71-7a8e-4e31-862b-283ca2b422d9/kill"}% 
```

try rebase?

**Comment by youngsofun** [comment]
https://github.com/sundy-li/databend/commits/fix-quoted

@sundy-li  the commit not include my pr

**Comment by BohuTANG** [comment]
@mergify update

**Comment by BohuTANG** [comment]
> log
> 
> > handler return 1 when expect bool
> 
> return `"true"` on main on my mac
> 
> and logic test can pass
> 
> ```
> (venv) ➜  logictest git:(fix) curl --request POST '127.0.0.1:8001/v1/query/' --header 'Content-Type: application/json' --data-raw '{"sql": "select true", "string_fields":true}' -u root:
> 
> {"id":"d601cd71-7a8e-4e31-862b-283ca2b422d9","session_id":"ea35c032-3b36-4cc4-a4b8-30445b9417d3","schema":{"fields":[{"name":"true","default_expr":null,"data_type":{"type":"Boolean"}}],"metadata":{}},"data":[["true"]],"state":"Succeeded","error":null,"stats":{"scan_progress":{"rows":1,"bytes":1},"running_time_ms":7.6482909999999995},"stats_uri":"/v1/query/d601cd71-7a8e-4e31-862b-283ca2b422d9","final_uri":"/v1/query/d601cd71-7a8e-4e31-862b-283ca2b422d9/final","next_uri":null,"kill_uri":"/v1/query/d601cd71-7a8e-4e31-862b-283ca2b422d9/kill"}% 
> ```
> 
> try rebase?

Comments `@mergify update` will do the rebase to main.

**Comment by youngsofun** [comment]
after rebase sundy`s pr

the 1/true is not failed 

but the `2 2 "2"-"2"` need to change,  seems to be fixed by this pr (originly store `\"2\"-\"2\"`?)

@sundy-li

**Comment by wubx** [comment]
nice

**Comment by youngsofun** [comment]
@sundy-li  logic test passed
stateless test fail:

```
03_0016_insert_into_values:                                             [ FAIL ] - result differs with:
--- /workspace/tests/suites/0_stateless/03_dml/03_0016_insert_into_values.result	2022-06-09 14:47:07.250882209 +0000
+++ /workspace/tests/suites/0_stateless/03_dml/03_0016_insert_into_values.stdout	2022-06-09 14:51:12.930006462 +0000
@@ -1,5 +1,5 @@
 -1	33	2021-08-15 10:00:00.000000	string1234
 101	67	2021-11-15 10:00:00.000000	string5678
 100	100
-a"b"c'd
 a"b"c\\'d
+a"b"c'd
```

**Pull request merged**